### PR TITLE
Combat is properly started when spell is cast and some more changes

### LINKED ISSUE #18877: [3.3.5] Taunt does not keep the target in combat [20$ bounty]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:** http://wotlk.evowow.com/?spell=355 does not keep the target in combat.

**Current behaviour:** (Tell us what happens.) ^

**Expected behaviour:** (Tell us what should happen instead.) Target should remain in combat while taunted, he is afterall, trying to attack you

**Steps to reproduce the problem:**

1. Cast taunt
2. Notice target not in combat
3. 

**Branch(es):** 3.3.5 / master (Tell us which branch(es) this issue affects.) 3.3.5

**TC rev. hash/commit:**   https://github.com/TrinityCore/TrinityCore/commit/439654d1e88dfc20ed437ac66e3b5a80833da773

**TDB version:**  62

**Operating system:**  ubuntu 14.04

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Issues addressed:** Closes #18877

**Tests performed:** Compiles, simple ingame tests

- Combat is properly started when spell is cast
- Combat is not started when Bind Sight aura is cast
- Taunting player properly puts both players in combat
- Increase PvP combat duration to 5500ms
- Set stand state to standing, when combat is started in MoveInLineOfSight
- Don't set passive controlled units in combat when owner starts combat
- Put owner in combat when pet really starts combat

**Review, tests, comments are welcomed**

### CODE REVIEW

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -3312,6 +3314,12 @@ void Spell::cast(bool skipCheck)
             m_caster->GetSpellHistory()->ResetCooldown(m_spellInfo->Id, true);
     }
 
+    // start combat at cast for delayed spells, only for explicit target
+    if (Unit* target = m_targets.GetUnitTarget())
+        if (m_caster->GetTypeId() == TYPEID_PLAYER || (m_caster->IsPet() && m_caster->IsControlledByPlayer()))
+            if (GetDelayMoment() > 0 && !m_caster->IsFriendlyTo(target) && !m_spellInfo->HasAura(SPELL_AURA_BIND_SIGHT) && (!m_spellInfo->IsPositive() || m_spellInfo->HasEffect(SPELL_EFFECT_DISPEL)))
+                m_caster->CombatStartOnCast(target, !m_spellInfo->HasAttribute(SPELL_ATTR3_NO_INITIAL_AGGRO), GetDelayMoment() + 500); // increase this time so we dont leave and enter combat in a moment
```
Comment (joschiwald) [inline src/server/game/Spells/Spell.cpp]: m_spellInfo->HasInitialAggro()

**Comment by Shauren** [comment]
Does this set combat flag (but not aggro) when missile is launched?
Also - Increase PvP combat duration to 5500ms - wot?

**Comment by xinef1** [comment]
1) Yes
2) It's hard to find reliable information now, but pvp combat was a bit greater than exactly 5 seconds

**Comment by robinsch** [comment]
In TBC it was clearly 5 seconds checked on a 1.5 second interval. You can check Woundman 1 (It's from the S2 ATR)

**Comment by xinef1** [comment]
Search for gouge + restealth

**Comment by Goatform** [comment]
@xinef1 is right. I can confirm that combat duration on current core is not correct. It is not 5.00 sec but instead (around) 5.50 sec some sources even show like 5.55 sec. I know this for a fact since I fixed this long time ago on my core, back in 2014.

**Comment by robinsch** [comment]
Mulitple Gouge + Stealths: https://www.youtube.com/watch?v=Dig13kEe40w
Combat Timer at 5 seconds: https://www.youtube.com/watch?v=6cMJFN-eNRU&t=623s
https://github.com/ccshiro/cc-buglist/issues/769

**Comment by Aokromes** [comment]
Conflicting files
src/server/game/Entities/Unit/Unit.cpp

**Comment by sirikfoll** [comment]
@Treeston Does your Combat/threat system rewrite overcome all the propositions of this PR? If not, could it be done? if it even is in the scope 😄

**Comment by Treeston** [comment]
Yes, this is fully superseded by #19930.

**Pull request closed without merging**